lab_3/main.py

```diff
@@ -20,7 +20,43 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
+    if not isinstance(text, str):
+        return ()
+    invaluable_trash = ['`', '~', '@', '#', '$', '%', '^', '&', '*', '(', ')', '-', '+', '=', '{', '[', ']', '}', '|',
+                        '\\', ':', ';', '"', "'", '<', ',', '>',
+                        '/', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0']
+    text = text.lower()
+    special_letters = ['ü', 'ö', 'ä', 'ß']
+    for symbols in invaluable_trash:
+        text = text.replace(symbols, '')
+    for letter in special_letters:
+        if letter == 'ü':
+            text = text.replace(letter, 'ue')
+        elif letter == 'ö':
+            text = text.replace(letter, 'oe')
+        elif letter == 'ä':
+            text = text.replace(letter, 'ae')
+        elif letter == 'ß':
```

Comment:
You can define a dictionary where the key would be a special character and the value would be the replacement.